app/src/main/java/com/ispc/gymapp/views/fragments/ExercisesBeginnerFragment.java

```diff
@@ -61,6 +88,50 @@ public void onCreate(Bundle savedInstanceState) {
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_exercises_beginner, container, false);
+        View view = inflater.inflate(R.layout.fragment_exercises_beginner, container, false);
+
+        progressBar = new ProgressBar(view.getContext());
+        progressBar.setVisibility(View.VISIBLE);
+
+        recyclerView = view.findViewById(R.id.recyclerExercises);
+        recyclerView.setHasFixedSize(true);
+        recyclerView.setLayoutManager(new LinearLayoutManager(view.getContext()));
+
+        db = FirebaseFirestore.getInstance();
+        exercises = new ArrayList<>();
+        exerciseListAdapter = new ExerciseListAdapter(view.getContext(), exercises);
+
+        recyclerView.setAdapter(exerciseListAdapter);
+
+        GetExercises();
+
+        return view;
+    }
+
+    private void GetExercises() {
+
+        db.collection("exercises").orderBy("title").addSnapshotListener(new EventListener<QuerySnapshot>() {
+            @Override
+            public void onEvent(@Nullable QuerySnapshot value, @Nullable FirebaseFirestoreException error) {
+
+                if (error != null) {
+                    Log.e("Firestore error", Objects.requireNonNull(error.getMessage()));
+                    return;
+                }
+
+                for (DocumentChange documentChange : value.getDocumentChanges()) {
+
+                    if (documentChange.getType() == DocumentChange.Type.ADDED) {
+                        exercises.add(documentChange.getDocument().toObject(Exercise.class));
+                    }
```

Comment:
```suggestion
                     if (documentChange.getType() == DocumentChange.Type.ADDED) {
                        Exercise exercise = documentChange.getDocument().toObject(Exercise.class);
                        if(exercise.getType().equals("beginner")){

                        exercises.add(exercise);
                        }
                    }
```

podrias hacer algo asi para filtrar en cada fragment y ponerle un enum o algo

Reply:
@federico42o leí tarde xd, lo hice de otra manera más rebuscada, creo, luego lo cambio de ultima.
Ahí subí los cambios, el RecyclerView ahora está en los 3 fragmentos y también le mandé algo para llevar al ExerciseDescription, donde usa los datos de la db también.

Reply:
quedo re bien igual, lo queres mergear?

Reply:
tambien podrias ver si es mejor pasarle ya la lista de cada ejercicio en el newInstance cuando creas el fragmento, y llamar a la bd una vez, o capaz es peor pero podrias probar q onda xd

Reply:
> tambien podrias ver si es mejor pasarle ya la lista de cada ejercicio en el newInstance cuando creas el fragmento, y llamar a la bd una vez, o capaz es peor pero podrias probar q onda xd

Podría, pero por ahora lo dejaré así, fue un dolor de cabeza hacer que funcione xD.

Si quieres lo mergeo, al menos probando no vi que se rompiera xd